pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -287,29 +288,33 @@ func (r *ReconcileKogitoApp) updateKogitoAppStatus(request *reconcile.Request, i
 }
 
 func (r *ReconcileKogitoApp) triggerBuilds(instance *v1alpha1.KogitoApp, bcs ...*obuildv1.BuildConfig) error {
-	bcMap := map[string][]obuildv1.BuildConfig{}
-	for _, bc := range bcs {
-		bcMap[bc.Labels[kogitores.LabelKeyBuildType]] = append(bcMap[bc.Labels[kogitores.LabelKeyBuildType]], *bc)
+	trigger := build.NewTrigger(r.client, instance.Name).
+		SelectOneBuildConfigWithLabel(map[string]string{
+			kogitores.LabelKeyBuildType:    string(kogitores.BuildTypeS2I),
+			kogitores.LabelKeyBuildVariant: string(kogitores.BuildVariantSource),
+		}, bcs...)
+	result, err := trigger.StartNewBuildIfNotRunning()
+	if err != nil {
+		return err
+	}
+	if result.Started {
+		log.Infof("Triggered build named %s for Kogito Application %s", result.BuildName, instance.Name)
 	}
 
-	// Trigger only the S2I builds, they will trigger runtime builds
-	if bcMap[string(kogitores.BuildTypeS2I)] != nil {
-		for _, bc := range bcMap[string(kogitores.BuildTypeS2I)] {
-			log.Infof("Buildconfigs are created, triggering build %s", bc.GetName())
-			if _, err := openshift.BuildConfigC(r.client).TriggerBuild(&bc, instance.Name); err != nil {
-				return err
-			}
+	// we don't have a s2i build, let's trigger the build source from image since it could be changed by an image stream
```

Comment:
The comment is quite confusing to me. Is this for binary builds? If so, then what "could be changed by an image stream"?

Reply:
If the image stream for the final runtime image is changed and we don't have `s2i` builds, we trigger this build.

I just ported the code from the old function. This will be refactored on https://issues.redhat.com/browse/KOGITO-1999

Reply:
I see, then please bear in mind https://issues.redhat.com/browse/KOGITO-2017 that runtime builds is missing image stream change trigger as @sutaakar pointed out.